cluster-controller/src/main/java/io/strimzi/controller/cluster/resources/TopicController.java

```diff
@@ -236,6 +254,7 @@ public static TopicController fromDeployment(String namespace, String cluster, D
             topicController.setReconciliationIntervalMs(vars.getOrDefault(KEY_FULL_RECONCILIATION_INTERVAL_MS, DEFAULT_FULL_RECONCILIATION_INTERVAL_MS));
             topicController.setZookeeperSessionTimeoutMs(vars.getOrDefault(KEY_ZOOKEEPER_SESSION_TIMEOUT_MS, DEFAULT_ZOOKEEPER_SESSION_TIMEOUT_MS));
             topicController.setTopicConfigMapLabels(vars.getOrDefault(KEY_CONFIGMAP_LABELS, defaultTopicConfigMapLabels(cluster)));
+            topicController.setTopicMetadataMaxAttempts(Integer.parseInt(vars.getOrDefault(KEY_TOPIC_METADATA_MAX_ATTEMPTS, String.valueOf(DEFAULT_TOPIC_METADATA_MAX_ATTEMPTS))));
```

Comment:
This is ugly: Converting an `int` to a `String` so we can use it as a default to be passed thru to `parseInt` so it gets passed back to an `int`! We must be able to extract a utility function for this.

Reply:
Do you mean an utility function for doing the same but "under the hood" ?
At same time it's also true the the `topicMetadataMaxAttempts` field is declared as `int` because it's really an integer but it's never used as a number but just for setting the env var for the topic controller. Maybe it could be declared as `String` as the other.

Reply:
Looking at this more closely I see it's in `fromDeployment()`. I think we need to be a bit careful here: It's stupid for the CC to fall over because someone changed the deployment to make this env var not be an int. So either we just pass around uninterpreted strings (so the CC cannot fall over) and leave it to the TC to validate its environment, or the CC needs to do the validation (but TC would still need to validate its environment).

To be honest a contract is forming between the CC and the TC and we need to decide at what point it makes sense for formalise that in the code. We might not be at that point yet, but we should be mindful that we might want to go that way in the future.